### PR TITLE
Initial web structure put together.

### PR DESCRIPTION
**Pull request by jesseobrien**
- Includes the ability to browse the site at locahost:8080
- Includes the react front-end working in /web
- Includes the HTTP web server out of internal/http
- Includes the self-signed certificates hardecoded
- Includes the ability to 'watch' files and rebuild the server on file
changes

### CODE REVIEW

File: .gitignore
Comment (awly) [inline .gitignore]: nit: include vendored dependencies in the repo, it's generally safer than pulling from the internet at compile time.

File: internal/http/http.go
Comment (awly) [inline internal/http/http.go]: since this is a library, return the error instead of stopping the process

File: go.mod
Comment (awly) [inline go.mod:7]: just curious: how did you end up choosing this framework?

File: internal/http/metrics.go
Comment (awly) [inline internal/http/metrics.go]: nit: more readable
```suggestion
		token := strings.TrimPrefix(authHeader, "Bearer ")
```

File: internal/http/metrics.go
Comment (awly) [inline internal/http/metrics.go]: this can open a timing attack, use a different comparison method

File: go.mod
Comment (jesseobrien) [reply go.mod:7]: Echo is the web server I have the most experience using. I'm always on the fence about being strongly idiomatic and wrapping say the `gorilla` bits around the `net/http` package. Echo provides "just idiomatic enough" and has a lot of nice little helpers for rendering JSON payloads, binding to structs, etc.

**Review by awly — APPROVED** [review]
LGTM but please wait for another approval for JS parts

**Review by russjones — APPROVED** [review]

**Review by alex-kovoy — APPROVED** [review]

**Pull request merged**
